applications/tari_base_node/src/parser.rs

```diff
@@ -556,6 +570,132 @@ impl Parser {
         });
     }
 
+    fn process_calc_timing<'a, I: Iterator<Item = &'a str>>(&self, args: I) {
+        let command_arg = args.take(4).collect::<Vec<&str>>();
+        if (command_arg.is_empty()) || (command_arg.len() > 2) {
+            println!("Command entered incorrectly, please use the following formats: ");
+            println!("calc-timing [first header height] [last header height]");
+            println!("calc-timing [amount of headers from top]");
+            return;
+        }
+        let height = if command_arg.len() == 2 {
+            let height = command_arg[1].parse::<u64>();
+            if height.is_err() {
+                println!("Invalid number provided");
+                return;
+            };
+            Some(height.unwrap())
+        } else {
+            None
+        };
+        let start = command_arg[0].parse::<u64>();
+        if start.is_err() {
+            println!("Invalid number provided");
+            return;
+        };
+        let counter = if command_arg.len() == 2 {
+            let start = start.unwrap();
+            let temp_height = height.clone().unwrap();
+            if temp_height <= start {
+                println!("start hight should be bigger than the end height");
+                return;
+            }
+            (temp_height - start) as usize
+        } else {
+            start.unwrap() as usize
+        };
+
```

Comment:
This can be pulled out into a function, since you're repeating the code in `process_list_headers`

Reply:
good point, forgot about this